### PR TITLE
Add Kafka OTEL storage exporter

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves https://github.com/jaegertracing/jaeger-opentelemetry-collector/issues/24

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,9 +88,13 @@ clean:
 	rm -rf cover.out .cover/ cover.html lint.log fmt.log
 
 .PHONY: test
-test: go-gen
+test: go-gen test-otel
 	bash -c "set -e; set -o pipefail; $(GOTEST) ./... | $(COLORIZE)"
 
+.PHONY: test-otel
+test-otel:
+	cd ${OTEL_COLLECTOR_DIR} && bash -c "set -e; set -o pipefail; $(GOTEST) ./... | $(COLORIZE)"
```
Comment (pavolloffay) [inline Makefile]: tests in `cmd/opentelemetry-collector` are excluded bc it is a separate go module. I have  found only this workaround.

It prints this when `go test ./..` is executed from the root dir

```
go: directory cmd/opentelemetry-collector is outside main module
go: directory cmd/opentelemetry-collector/app/defaults is outside main module
go: directory cmd/opentelemetry-collector/app/exporter is outside main module
go: directory cmd/opentelemetry-collector/app/exporter/kafka is outside main mo
```

File: cmd/opentelemetry-collector/app/exporter/kafka/exporter.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+// New creates new Kafka exporter
+func New(config *Config, log *zap.Logger) (exporter.TraceExporter, error) {
+	f := kafka.NewFactory()
+	f.InitFromOptions(config.Options)
+	err := f.Initialize(metrics.NullFactory, log)
+	if err != nil {
+		return nil, err
+	}
+	return create(f, config)
```
Comment (pavolloffay) [inline cmd/opentelemetry-collector/app/exporter/kafka/exporter.go]: Only this line is not being covered from the whole PR. I cannot pass the initialization, there is no way to change the objects in the Kafka config for mocked builders.

File: cmd/opentelemetry-collector/app/exporter/kafka/factory_test.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	factory := &Factory{OptionsFactory: func() *kafka.Options {
+		opts := DefaultOptions()
+		opts.InitFromViper(v)
+		return opts
+	}}
+	exporter, err := factory.CreateTraceExporter(zap.NewNop(), factory.CreateDefaultConfig())
+	require.Nil(t, exporter)
+	assert.EqualError(t, err, "kafka: client has run out of available brokers to talk to (Is your cluster reachable?)")
+
+	exporter, err = factory.CreateTraceExporter(zap.NewNop(), nil)
```
Comment (pavolloffay) [inline cmd/opentelemetry-collector/app/exporter/kafka/factory_test.go]: TODO move this to a separate test

File: pkg/kafka/auth/kerberos.go [hunk 4]
```diff
@@ -20,13 +20,13 @@ import (
...
-	Password    string
-	ConfigPath  string
-	KeyTabPath  string
+	ServiceName string `mapstructure:"service_name"`
+	Realm       string `mapstructure:"realm"`
+	UseKeyTab   bool   `mapstructure:"use_key_tab"`
+	Username    string `mapstructure:"username"`
+	Password    string `mapstructure:"password"`
+	ConfigPath  string `mapstructure:"config_file"`
+	KeyTabPath  string `mapstructure:"key_tab_file"`
```
Comment (objectiser) [inline pkg/kafka/auth/kerberos.go]: `keytab_file`?

File: pkg/kafka/auth/kerberos.go [hunk 5]
```diff
@@ -20,13 +20,13 @@ import (
...
-	ServiceName string
-	Realm       string
-	UseKeyTab   bool
-	Username    string
-	Password    string
-	ConfigPath  string
-	KeyTabPath  string
+	ServiceName string `mapstructure:"service_name"`
+	Realm       string `mapstructure:"realm"`
+	UseKeyTab   bool   `mapstructure:"use_key_tab"`
```
Comment (objectiser) [inline pkg/kafka/auth/kerberos.go]: `use_keytab`?

File: plugin/storage/kafka/options.go [hunk 6]
```diff
@@ -108,9 +108,9 @@ var (
 
 // Options stores the configuration options for Kafka
 type Options struct {
-	config   producer.Configuration
-	topic    string
-	encoding string
+	Config   producer.Configuration `mapstructure:",squash"`
```
Comment (objectiser) [inline plugin/storage/kafka/options.go]: Does this need `mapstructure:"producer....`?

Comment (pavolloffay) [reply plugin/storage/kafka/options.go]: I was thinking about it, but given it is scoped under exporters it does not make much sense.

File: cmd/opentelemetry-collector/app/exporter/kafka/exporter.go (hunk 2 above)
Comment (pavolloffay) [reply cmd/opentelemetry-collector/app/exporter/kafka/exporter.go]: @yurishkuro any idea how to fix the coverage? Even though only a single line is not covered the total is: 

```
ok  	github.com/jaegertracing/jaeger/cmd/opentelemetry-collector/app/exporter/kafka	0.767s	coverage: 94.7% of statements

```

**Comment by pavolloffay** [comment]
Travis does not want to report the result. hence reopening the PR

**Comment by pavolloffay** [comment]
@yurishkuro feel free to comment. I will fix anything in a follow-up PR.

**Pull request merged**